app/validators/form_dob_validator.rb

```diff
@@ -0,0 +1,52 @@
+module FormDobValidator
+  extend ActiveSupport::Concern
+  include ActionView::Helpers::TranslationHelper
+
+  included do
+    include ActiveModel::Validations::Callbacks
+    validate :validate_dob
+  end
+
+  private
+
+  def validate_dob
+    return unless dob.present?
+    unless meet_min_age_req?(dob)
+      errors.add(:dob, message: dob_min_age_error, type: :dob_min_age_error)
+    end
+  end
+
+  # @param [String|Date|ActionController::Parameters] dob
+  # @return [Boolean] false unless it meets the required minimal age,
+  #  including invalid input value and types
+  def meet_min_age_req?(dob)
+    return false if dob.nil?
+    unless dob.is_a?(String) || dob.is_a?(ActionController::Parameters) ||
+           dob.is_a?(Date)
+      return false
+    end
```

Comment:
The rails controller in our case state_id step, that's pass in ActionController::Parameters. A little of broad on the type of input just in case it(the validator) used in various scenario, personally I dont like the name form_dob_validator, this part of logic can exists out of the MVC framework(extracted out), let's say as a utility used by other non UI portion.

Reply:
refactored.